core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -117,16 +117,18 @@ class RangePolicy : public Impl::PolicyTraits<Properties...> {
   inline RangePolicy(const typename traits::execution_space& work_space,
                      const member_type work_begin, const member_type work_end)
       : m_space(work_space),
-        m_begin(work_begin < work_end ? work_begin : 0),
-        m_end(work_begin < work_end ? work_end : 0),
+        m_begin(work_begin),
+        m_end(work_end),
         m_granularity(0),
         m_granularity_mask(0) {
+    check_bounds_validity();
     set_auto_chunk_size();
   }
 
   /** \brief  Total range */
   inline RangePolicy(const member_type work_begin, const member_type work_end)
       : RangePolicy(typename traits::execution_space(), work_begin, work_end) {
+    check_bounds_validity();
```

Comment:
You'd never get here since the construction is deferred to the other `RangePolicy` constructor that would error out.

Reply:
I did notice it before, but I placed it in just to have this check logic be paired with each `set_auto_chunk_size()` call, which makes uses of `m_begin` and `m_end`.
If you feel strongly that we don't want this in this constructor (and the other similar constructor below), I'll remove the lines.

Reply:
I can live with it.  There are things that bother me more in that file :)